components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -118,10 +118,12 @@ bool IsVettedSearchEngine(const std::string& host) {
     size_t found_pos = host.find((*g_vetted_search_engines)[i]);
     if (found_pos != std::string::npos) {
       size_t last_dot_pos = host.find(".", found_pos + 1);
-      if (last_dot_pos == std::string::npos) {
+      if (last_dot_pos == std::string::npos)
         return false;
-      }
-      if (host.find(".", last_dot_pos + 1) == std::string::npos) {
+      if (host.find(".", last_dot_pos + 1) == std::string::npos &&
+          (found_pos == 0 || host[found_pos - 1] == '/' ||
+           host[found_pos - 1] == '\\' || host[found_pos - 1] == ':' ||
+           host[found_pos - 1] == '.')) {
```

Comment:
@pes10k do we want this logic to apply to subdomains as well?

Reply:
Yes, full host name only.  We don't want to, say, exempt, mail.google.com from cosmetic filtering

Reply:
@antonok-edm @pes10k corrected